components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc

```diff
@@ -274,13 +279,46 @@ bool CosmeticFiltersJSHandler::OnIsFirstParty(const std::string& url_string) {
 void CosmeticFiltersJSHandler::OnAddSiteCosmeticFilter(
     const std::string& selector) {
   const auto host = url_.host();
-  auto handler = MakeCosmeticFiltersHandler(render_frame_);
-  handler->AddSiteCosmeticFilter(selector);
+  GetElementPickerRemoteHandler()->AddSiteCosmeticFilter(selector);
 }
 
 void CosmeticFiltersJSHandler::OnManageCustomFilters() {
-  auto handler = MakeCosmeticFiltersHandler(render_frame_);
-  handler->ManageCustomFilters();
+  GetElementPickerRemoteHandler()->ManageCustomFilters();
+}
+
+mojo::AssociatedRemote<cosmetic_filters::mojom::CosmeticFiltersHandler>&
+CosmeticFiltersJSHandler::GetElementPickerRemoteHandler() {
+  if (!element_picker_actions_handler_ ||
```

Comment:
Kooks like we should just check `is_connected()`:
https://source.chromium.org/chromium/chromium/src/+/main:mojo/public/cpp/bindings/associated_remote.h;l=94-114